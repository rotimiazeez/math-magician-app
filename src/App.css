* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-color: #fff;
}

.container {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  margin: 0 auto;
  width: 341px;
  text-align: center;
  background-color: #e0e0e0;
  border: 1px solid #c5c5c5;
}

.keypad {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: minmax(60px, auto);
}

input[type="text"] {
  height: 75px;
  width: 341px;
  text-align: right;
  background: #858694;
  border: 1px solid #c5c5c5;
  color: #fff;
  font-size: 40px;
}

.highlight {
  background-color: #f5913e;
}

.bottom {
  position: absolute;
  top: 100%;
  height: 60px;
  width: 170px;
}

.bottomII,
.bottomIII {
  position: absolute;
  top: 100%;
  height: 60px;
  left: 171px;
  width: 85px;
}

.bottomIII {
  left: 254px;
}
