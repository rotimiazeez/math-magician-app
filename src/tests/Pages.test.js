import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from '../App';

describe('pages test snapshots', () => {
  it('testing home page', () => {
    render(<App />);
    const Home = screen.getByText('Home');
    fireEvent.click(Home);
    expect(screen.getByText('Welcome to our page')).toMatchSnapshot();
  });
  it('testing Quote page', () => {
    render(<App />);
    const Quote = screen.getByText('Quotes');
    fireEvent.click(Quote);
    expect(screen.getByText('"We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics". â€” Katherine Johnson, African-American mathematician')).toMatchSnapshot();
  });
  it('testing Calculator page', () => {
    render(<App />);
    const Calculator = screen.getByText('Calculator');
    fireEvent.click(Calculator);
    expect(screen.getByText('0')).toMatchSnapshot();
  });
});
