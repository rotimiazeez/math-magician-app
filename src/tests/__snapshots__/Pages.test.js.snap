// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages test snapshots testing Calculator page 1`] = `
<button
  class="bottom"
  name="0"
  type="button"
>
  0
</button>
`;

exports[`pages test snapshots testing Quote page 1`] = `
<p>
  "We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics". â€” Katherine Johnson, African-American mathematician
</p>
`;

exports[`pages test snapshots testing home page 1`] = `
<h4>
   Welcome to our page
</h4>
`;
